### PR TITLE
Added Circular Linked List Data Structure

### PR DESCRIPTION
**Pull request by mohit-gogitter**
### Circular Linked List
A **Circular Linked List** Data Structure is a variation of the linked list where the last node points back to the first node, forming a circular structure. In this implementation, I've created a circular linked list where each node contains data and a reference to the next node.

Here's a summary of key operations and their time complexities in this implementation:

1. **Insert at the Beginning**: Adds a new node at the start of the list and links the new node to the existing nodes, ensuring the circular connection is maintained. (Time Complexity: O(1))
2. **Insert at the End**: Adds a new node at the end of the list by finding the last node and updating its pointer to the new node, which is then linked back to the first node. (Time Complexity: O(n))
3. **Insert After a Given Node**: Searches for a specific node in the list and inserts a new node after it while maintaining the circular connection. (Time Complexity: O(n))
4. **Delete a Node**: Removes a node from the list by updating the pointers of the surrounding nodes. (Time Complexity: O(n))

**Where It Can Be Used:**
- Operating Systems: In process scheduling where tasks are selected in a circular fashion.
- Multimedia Applications: To manage looping playlists for audio and video playback.
- Networking: In token-passing protocols for network communication.
- Simulation or Game Development: For player turns in multiplayer games, simulating continuous cycles of events.

**CheckList**
- [X] I have performed a self-review of my code
- [X] My code follows the style guidelines of this project
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Comments in areas I changed are up to date
- [X] I have added comments to hard-to-understand areas of my code
- [X] I have made corresponding changes to the README.md

### CODE REVIEW

File: DataStructures/LinkedList/CircularLinkedList/CircularLinkedList.cs [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+        /// <summary>
+        /// Inserts a new node after a specific value in the list.
+        /// </summary>
+        /// <param name="value">The value to insert the node after.</param>
+        /// <param name="data">The data to insert into the new node.</param>
+        public void InsertAfter(T value, T data)
+        {
+            if (IsEmpty())
+            {
+                Console.WriteLine("List is empty.");
```
Comment (siriak) [inline DataStructures/LinkedList/CircularLinkedList/CircularLinkedList.cs:82]: Please replace console output with exceptions

**Review by siriak — CHANGES_REQUESTED** [review]

File: DataStructures/LinkedList/CircularLinkedList/CircularLinkedList.cs (hunk 1 above)
Comment (mohit-gogitter) [reply DataStructures/LinkedList/CircularLinkedList/CircularLinkedList.cs]: Done!

File: DataStructures/LinkedList/CircularLinkedList/CircularLinkedList.cs [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+                previous = current;
+                current = current.Next;
+            }
+            while (current != tail!.Next);
+        }
+
+        /// <summary>
+        /// Displays the contents of the Circular Linked List.
+        /// </summary>
+        public void Display()
```
Comment (siriak) [inline DataStructures/LinkedList/CircularLinkedList/CircularLinkedList.cs:155]: Please remove display method and display-related logic from tests

Comment (mohit-gogitter) [reply DataStructures/LinkedList/CircularLinkedList/CircularLinkedList.cs]: Done

**Comment by mohit-gogitter** [comment]
@siriak Can you please review

**Comment by mohit-gogitter** [comment]
@siriak Can you please review as the PR is opened since 2 weeks

**Review by siriak — APPROVED** [review]
Looks good, thanks!

**Pull request merged**